buildkitd/oom-adjust.sh.template

```diff
@@ -0,0 +1,29 @@
+#! /bin/sh
+
+set -e
+
+if [ "${OOM_SCORE_ADJ}" -eq "0" ]; then
+    exit 0
+fi
+
+for PID in $(pidof buildkit-runc)
+do
+    PID_NAME=$(cat /proc/$PID/cmdline)
+    ! ${BUILDKIT_DEBUG} || echo "$(date) | Checking $PID($PID_NAME)..." >> /var/log/oom_adj
+
+    if [ "$PID_NAME" = "buildkit-runcinit" ]; then
+        ! ${BUILDKIT_DEBUG} || echo "$(date) |   $PID was buildkit-runcinit, ignoring" >> /var/log/oom_adj
+        continue
+    fi
+
+    childpids=$(pgrep -P $PID | xargs)
+    for CHILD_PID in $(pgrep -P $PID)
+    do
+        ! ${BUILDKIT_DEBUG} || echo "$(date) |   $PID has child: $CHILD_PID($(cat /proc/$CHILD_PID/cmdline)) with oom_score_adj: $(/proc/$CHILD_PID/oom_score_adj)" >> /var/log/oom_adj
+    done
+
+    ! ${BUILDKIT_DEBUG} || echo "$(date) |   Adjusting pid $PID oom_score_adj to ${OOM_SCORE_ADJ}" >> /var/log/oom_adj
+    ! ${BUILDKIT_DEBUG} || echo "${OOM_SCORE_ADJ}" > /proc/$PID/oom_score_adj
+done
+
+! ${BUILDKIT_DEBUG} || echo "$(date) | --" >> /var/log/oom_adj
```

Comment:
I *think* that if we quote `"${BUILDKIT_DEBUG}"` (this shouldn't change the functionality), we can add this template to the shellcheck list in our main Earthfile.

Reply:
Ill look and see what shellcheck wants and if it affects anything

Reply:
Shellcheck is happy with it without quoting these

Reply:
It's usually a good idea to quote for style mainly - in case it ever becomes a variable that contains spaces. I think shellcheck ignores this rule in some operations.

In any case - all looks good from my point of view!

Reply:
Quoted